client/catalog/src/app/datatypes/DatatypeNew.svelte

```diff
@@ -0,0 +1,117 @@
+<script>
+import client from '../../data/client.js';
+import {restricts} from '../util.js';
+import { breadcrumbStore } from '../stores.js';
+import { onMount } from 'svelte';
+import { Storage } from '../../data/enums.js';
+import HLButton from '../../ui/HLButton.svelte';
+import HLRowButton from '../../ui/HLRowButton.svelte';
+import HLRow from '../../ui/HLRow.svelte';
+import HLTextBig from '../../ui/HLTextBig.svelte';
+import HLSelect from '../../ui/HLSelect.svelte';
+import HLTable from '../../ui/HLTable.svelte';
+import CodeMirror from 'codemirror';
+import 'codemirror/mode/python/python.js';
+
+breadcrumbStore.set(
+	[{
+		href: "/datatypes",
+		text: "Datatypes",
+	}, {
+		href: "/Datatpes/new",
+		text: "New",
+	}]
+);
+
+const newDatatypeOp = {
+	name: "",
+	storedAs: 0,
+	validator : {
+		create : {
+			name : "",
+			runtime: 2,
+			code: "",
+			functionSignature: 0,
+		}
+	}
+}
+
+var cm;
+onMount(async () => {
+	cm = CodeMirror.fromTextArea(document.querySelector('#code'), {
+		mode: {name:"python"},
+		lineNumbers: true,
+		indentUnit: 4,
+		theme: "duotone-dark",
+	});
+	cm.setValue(
+`# Compile Regular Expression for valid US Phone Numbers
```

Comment:
I wonder if we might copy workshop in having a blank textarea but then some kind of tutorial text/example to the right ofit. Definitely OK to punt on that piece if you wanna come back to it later

Reply:
Gonna put now